client/app/api/course/ExperiencePointsRecord.ts

```diff
@@ -11,45 +13,57 @@ export default class ExperiencePointsRecordAPI extends BaseCourseAPI {
     return `/courses/${this.courseId}`;
   }
 
+  /**
+   * Fetches all experience points records from all user
+   */
+  indexAll(
+    studentId?: number,
+    pageNum: number = 1,
+  ): Promise<AxiosResponse<ExperiencePointsRecords>> {
+    return this.client.get(`${this.#urlPrefix}/experience_points_records`, {
+      params: { 'filter[page_num]': pageNum, 'filter[student_id]': studentId },
+    });
+  }
+
   /**
    * Fetches all experience points records from a user
    */
   index(
     userId: number,
     pageNum: number = 1,
-  ): Promise<
-    AxiosResponse<{
-      courseUserName: string;
-      rowCount: number;
-      experiencePointRecords: ExperiencePointsRecordListData[];
-    }>
-  > {
+  ): Promise<AxiosResponse<ExperiencePointsRecords>> {
     return this.client.get(
-      `${
-        this.#urlPrefix
-      }/users/${userId}/experience_points_records?filter[page_num]=${pageNum}`,
+      `${this.#urlPrefix}/users/${userId}/experience_points_records`,
+      { params: { 'filter[page_num]': pageNum } },
     );
   }
 
+  download(studentId?: number): Promise<AxiosResponse<JobSubmitted>> {
+    return this.client.get(`${this.#urlPrefix}/download_experience_points`, {
+      params: { 'filter[student_id]': studentId },
+    });
+  }
+
   /**
    * Update an experience points record for a user
    */
   update(
     params: UpdateExperiencePointsRecordPatchData,
     recordId: number,
+    studentId?: number,
   ): Promise<AxiosResponse<ExperiencePointsRecordListData>> {
     const url = `${this.#urlPrefix}/users/${
-      this.courseUserId
+      this.courseUserId ?? studentId
```

Comment:
I suggest to not use `this.courseUserId` at all so `studentId` is not optional anymore. In this case, we dont try to infer the api path from existing url. Same for delete below.

So, `updateExperiencePointsRecord` and `deleteExperiencePointsRecord` also require `studentId`.
Instead of `studentId`, call it `courseUserId` to be more generic.

Reply:
I'm not quite sure actually that courseUserId is more intuitive than studentId. Indeed, the url formed has prefixes of users, but in here, the owner of the exp points are student itself

The reason I need to put emphasis on student is because inside the entity of exp points, there is another object similar with student as well (updater) and thus I need to differentiate if the id mentioned belongs to student or updater